wandb/sdk/launch/builder/kaniko_builder.py

```diff
@@ -263,8 +263,11 @@ async def build_image(
         repo_uri = await self.registry.get_repo_uri()
         image_uri = repo_uri + ":" + image_tag
 
+        # If DOCKER_CONFIG_SECRET is set then we can't use the registry object
+        # to check whether the image exists.
```

Comment:
Why not? Shouldn't we be able to?

Reply:
If the registry handler is being used we expect to have credentials for the azure/aws/gcp sdk, which we do not. And if it's an anonymous registry then we can't do that. We could move towards relying more on the dockerconfig for the agent being set and having agent just use the docker cli to check if images exist. I will touch up the comment

Reply:
Yeah, this seems like a loss of functionality, if users set up the KanikoConfigSecret than they won't reuse images and will build everytime, which is not ideal. If we can use docker directly to check if it exists that would be preferred

Reply:
Are you blocking your stamp of this PR on this? I believe this PR strictly adds functionality and I think getting docker manifest inspect to work in the agent running in kubernetes is out of scope

Reply:
That will also require a helm chart mount for docker credentials and/or CLI helpers for docker creds and a docker cli being baked into the agent image

Reply:
No, just making sure I understand. So this only happens in the case of an out of cloud k8s cluster relying on the KanikoDockerConfigSecret, which *did not work* previously, but now will. However, the downside is, we will always build images even if they already exist.

We should add a log line indicating that we are skipping the check (so when a user runs into this we have an easy way of determining why). And resolving the problem